daft/io/_range.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+from daft import DataType
+from daft.io._generator import GeneratorScanOperator
+from daft.logical.schema import Schema
+from daft.table.table import Table
+
+
+def _range_generators(start: int, end: int, step: int, partitions: int) -> Iterator[Callable[[], Iterator[Table]]]:
+    def generator_for_partition(partition_idx: int) -> Callable[[], Iterator[Table]]:
+        def generator() -> Iterator[Table]:
```

Comment:
1. It would be cleaner to use `functools.partial` here instead of having the nested functions. Example:
```
for partition_idx in range(partitions):
    yield partial(generator, partition_idx)
```
2. It would also be cleaner to calculate all the partition bounds upfront.
3. I think there will be some issues in this implementation when there are partitions, specifically to do with balanced partitions and the step. Can you add  a TODO in the comments that partitioning with range scan is currently untested and unused.